File: Main/Source/GlobalSwapShop_sln/Source/GlobalSwapShop_web/FVK/Tools/FVKConfig.cs
// WARNING: 
// The source code of this class can be used and modified to your particular needs. 
// Sharing the class with unregistered users or using it with unregistered websites 
// or Facebook applications is not allowed and violates the rights of intellectual 
// property. 

using System;
using System.Data;
using System.Configuration;
using System.Linq;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.HtmlControls;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Xml.Linq;

namespace FVK
{
    public static class FVKConfig
    {
        public static string AppId { get { return appId; } set { appId = value; } }
        public static string AppSecret { get { return appSecret; } set { appSecret = value; } }
        public static string AppName { get { return appName; } set { appName = value; } }
        public static string AppUrl { get { return appUrl; } set { appUrl = value; } }

        static FVKConfig()
        {
            // Initialization
            appId = ConfigurationManager.AppSettings["appId"];
            appSecret = ConfigurationManager.AppSettings["appSecret"];
            appName = ConfigurationManager.AppSettings["appName"];
            appUrl = ConfigurationManager.AppSettings["appUrl"];

            // checking
            if (appId == null || appId == "")
            {
                throw new Exception("AppId is not defined in Web.config");
            }

            if (appSecret == null || appSecret == "")
            {
                throw new Exception("Secret is not defined in Web.config");
            }

            if (appName == null || appName == "")
            {
                throw new Exception("AppName is not defined in Web.config");
            }

            if (appUrl == null || appUrl == "")
            {
                throw new Exception("AppCanvasUrl is not defined in Web.config");
            }
            else
            {
                if (!appUrl.EndsWith("/"))
                {
                    appUrl += "/";
                }
            }
        }

        private static string appId;
        private static string appSecret;
        private static string appName;
        private static string appUrl;
    }
}

File: Main/ReferenceDLLs/Pack1/Facebook Visual Kit VB/FVK Source Code/FVK/FVK/Tools/FVKConfig.vb
' WARNING: 
' The source code of this class can be used and modified to your particular needs. 
' Sharing the class with unregistered users or using it with unregistered websites 
' or Facebook applications is not allowed and violates the rights of intellectual 
' property. 

Imports System.Data
Imports System.Configuration
Imports System.Linq
Imports System.Web
Imports System.Web.Security
Imports System.Web.UI
Imports System.Web.UI.HtmlControls
Imports System.Web.UI.WebControls
Imports System.Web.UI.WebControls.WebParts
Imports System.Xml.Linq

Namespace FVK
    Public NotInheritable Class FVKConfig
        Private Sub New()
        End Sub
        Public Shared Property AppId() As String
            Get
                Return m_appId
            End Get
            Set(ByVal value As String)
                m_appId = value
            End Set
        End Property
        Public Shared Property AppSecret() As String
            Get
                Return m_appSecret
            End Get
            Set(ByVal value As String)
                m_appSecret = value
            End Set
        End Property
        Public Shared Property AppName() As String
            Get
                Return m_appName
            End Get
            Set(ByVal value As String)
                m_appName = value
            End Set
        End Property
        Public Shared Property AppUrl() As String
            Get
                Return m_appUrl
            End Get
            Set(ByVal value As String)
                m_appUrl = value
            End Set
        End Property

        Shared Sub New()
            ' Initialization
            m_appId = ConfigurationManager.AppSettings("appId")
            m_appSecret = ConfigurationManager.AppSettings("appSecret")
            m_appName = ConfigurationManager.AppSettings("appName")
            m_appUrl = ConfigurationManager.AppSettings("appUrl")

            ' checking
            If m_appId Is Nothing OrElse m_appId = "" Then
                Throw New Exception("AppId is not defined in Web.config")
            End If

            If m_appSecret Is Nothing OrElse m_appSecret = "" Then
                Throw New Exception("Secret is not defined in Web.config")
            End If

            If m_appName Is Nothing OrElse m_appName = "" Then
                Throw New Exception("AppName is not defined in Web.config")
            End If

            If m_appUrl Is Nothing OrElse m_appUrl = "" Then
                Throw New Exception("AppCanvasUrl is not defined in Web.config")
            Else
                If Not m_appUrl.EndsWith("/") Then
                    m_appUrl += "/"
                End If
            End If
        End Sub

        Private Shared m_appId As String
        Private Shared m_appSecret As String
        Private Shared m_appName As String
        Private Shared m_appUrl As String
    End Class
End Namespace
